src/prefect/types/__init__.py

```diff
@@ -200,6 +200,25 @@ def convert_none_to_empty_dict(v: Optional[KeyValueLabels]) -> KeyValueLabels:
 ]
 
 
+def deserialize_dict_if_string(value: Any) -> dict[str, Any]:
+    """Useful when a value is sometimes passed as a string or natively as a dict.
+
+    Example, setting the custom headers via the CLI involves passing a string format of a dict:
+    ```
+    prefect settings set PREFECT_CLIENT_CUSTOM_HEADERS='{"X-Test-Header": "test-value", "Authorization": "Bearer token123"}'
+    ```
+    """
+    if isinstance(value, str):
+        return orjson.loads(value)
+    return value
+
+
+SometimesSerializedDict = Annotated[
```

Comment:
Naming is difficult, but I think this name captures the purpose of this type
```suggestion
JsonStringOrDict = Annotated[
```